test/utils/executor.hpp

```diff
@@ -66,92 +64,41 @@ void init_executor(std::shared_ptr<gko::ReferenceExecutor> ref,
                    std::shared_ptr<gko::CudaExecutor>& exec)
 {
     ASSERT_GT(gko::CudaExecutor::get_num_devices(), 0);
-    exec = gko::CudaExecutor::create(0, ref);
+    exec = gko::CudaExecutor::create(
```

Comment:
I think it would be better if we had MPI-specific test utils, otherwise running the tests manually might have unexpected side-effects